backend/ethereum/wallet/simple/wallet.go

```diff
@@ -59,17 +69,19 @@ func (w *Wallet) NewRandomAccount(prng *rand.Rand) wallet.Account {
 	if err != nil {
 		log.Panicf("Creating account: %v", err)
 	}
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
 
 	addr := crypto.PubkeyToAddress(privateKey.PublicKey)
-	if !w.Contains(addr) {
+	if !w.contains(addr) {
```

Comment:
If your prng is fine, this should never happen.

The current behavior is not plausible to me. If you want to make sure that a new account is created, you should either retry key generation or simply panic in case there is already an account for the generated key.